index.js

```diff
@@ -298,68 +324,88 @@ function cmpType (a, b) {
   return typeOrder[a.type] - typeOrder[b.type]
 }
 
-DB.prototype._onpt = function (pt, seen, cb) {
+// Given a node by its version, this collects the node itself, and also
+// recursively climbs all ways and relations that the node (or its referers)
+// are referred to by.
+// OsmVersion, { OsmVersion: Boolean } -> [OsmDocument]
+DB.prototype._collectNodeAndReferers = function (version, seenAccum, cb) {
   cb = once(cb || noop)
   var self = this
-  var link = pt.value.toString('hex')
-  if (has(seen, link)) return cb(null, [])
-  seen[link] = true
+  if (has(seenAccum, version)) return cb(null, [])
+  seenAccum[version] = true
   var res = [], added = {}, pending = 2
-  self.log.get(link, function (err, doc) {
-    if (doc && doc.value && doc.value.k && doc.value.v) {
-      addDoc(doc.value.k, link, doc.value.v)
-    }
+
+  self.log.get(version, function (err, node) {
+    addDocFromNode(node)
     if (--pending === 0) cb(null, res)
   })
-  self._links(link, function onlinks (err, links) {
+
+  self._getReferers(version, function onlinks (err, links) {
```

Comment:
Let's de-inline this function so we can just `self._getReferers(version, onlinks)` here and below